iceoryx_hoofs/test/moduletests/test_relative_pointer.cpp

```diff
@@ -86,72 +96,88 @@ TYPED_TEST_SUITE(RelativePointer_test, Types);
+                                               static_cast<void*>(this->memoryPartition[0]),
+                                               SHARED_MEMORY_SIZE),
+              true);
+    EXPECT_EQ(BaseRelativePointer::registerPtr(iox::rp::BaseRelativePointer::id_t{2U},
+                                               static_cast<void*>(this->memoryPartition[1]),
+                                               SHARED_MEMORY_SIZE),
+              true);
+
+    auto* ptr0 = this->partitionPtr(0U);
+    auto* ptr1 = this->partitionPtr(1U);
 
     {
-        auto offset = SHARED_MEMORY_SIZE / 2;
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
+        auto offset = SHARED_MEMORY_SIZE / 2U;
+        // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+        auto* typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
 
         RelativePointer<TypeParam> rp;
         rp = typedPtr;
         EXPECT_EQ(rp.getOffset(), offset);
-        EXPECT_EQ(rp.getId(), 1);
+        EXPECT_EQ(rp.getId(), 1U);
         EXPECT_NE(rp, nullptr);
     }
 
     {
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr0);
+        auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(ptr0));
         RelativePointer<TypeParam> rp(typedPtr);
         EXPECT_EQ(rp.getOffset(), 0);
-        EXPECT_EQ(rp.getId(), 1);
+        EXPECT_EQ(rp.getId(), 1U);
         EXPECT_NE(rp, nullptr);
     }
 
     {
-        auto offset = SHARED_MEMORY_SIZE / 2;
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
+        auto offset = SHARED_MEMORY_SIZE / 2U;
+        // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+        auto* typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
         RelativePointer<TypeParam> rp(typedPtr);
         EXPECT_EQ(rp.getOffset(), offset);
-        EXPECT_EQ(rp.getId(), 1);
+        EXPECT_EQ(rp.getId(), 1U);
         EXPECT_NE(rp, nullptr);
     }
 
     {
-        auto offset = SHARED_MEMORY_SIZE - 1;
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
+        auto offset = SHARED_MEMORY_SIZE - 1U;
+        // Pointer arithmetic needed for tests
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+        auto* typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
         RelativePointer<TypeParam> rp(typedPtr);
         EXPECT_EQ(rp.getOffset(), offset);
-        EXPECT_EQ(rp.getId(), 1);
+        EXPECT_EQ(rp.getId(), 1U);
         EXPECT_NE(rp, nullptr);
     }
 
     {
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr1);
+        auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(ptr1));
```

Comment:
please use reinterpret_cast

Reply:
`static_cast` has several advantages e.g. it can be used during compile-time, see [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1521#discussion_r936463947).